flink-runtime/src/test/java/org/apache/flink/runtime/io/network/partition/hybrid/TestingSpillingInfoProvider.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.io.network.partition.hybrid;
+
+import java.util.ArrayDeque;
+import java.util.ArrayList;
+import java.util.Deque;
+import java.util.List;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+/** Mock {@link HsSpillingInfoProvider} for test. */
+public class TestingSpillingInfoProvider implements HsSpillingInfoProvider {
+    private final int numSubpartitions;
+
+    private Supplier<List<Integer>> consumptionProgressSupplier = ArrayList::new;
+
+    private Function<Integer, Deque<BufferWithIdentity>> unSpilledBuffersFunction =
+            (ignore) -> new ArrayDeque<>();
+
+    private Supplier<Integer> numTotalUnSpilledBuffersSupplier = () -> 0;
+
+    private Supplier<Integer> currentPoolSizeSupplier = () -> 0;
```

Comment:
These should be final. They can be decided via a builder/factory. Once created, behaviors of the testing provider should be deterministic.

Reply:
changed to create by builder